### DISCUSSION TITLE
Zero-and-One-Adjusted Beta

### DISCUSSION THREAD

**Discussion by neverfox** [StatMixedML/LightGBMLSS#20] [Ideas]
Very cool to see the addition of the zero-adjusted/inflated distributions. Would it be practically possible to have a zero-_and_-one-adjusted/inflated Beta, i.e. [ZOIB](https://journal.r-project.org/archive/2015/RJ-2015-019/), a Beta that allows the response to be [0, 1]? This goes towards the idea of using it with binary classification responses and getting, instead of the usual, natural p parameter of a Bernoulli, the distribution of p itself as a Beta distribution. This would be interesting to ask probabilistic questions _about_ the predicted probability, I think. I could probably hack it by modifying the data with a small epsilon, but it ZABeta opens the door to doing that in a principled way. I think the idea would mean:

* A gate/probability, given by a Bernoulii, of a 0 or 1
* A gate/probability, given by a Bernoulii, of a 0 (or 1, if you like) given that it's 0 or 1
* The beta that models the outcome between 0 and 1 if it’s not zero or not one

**Comment by StatMixedML** [comment]
Thanks for your detailed outline.

Extending the current Beta distribution from (0,1) to 

- ZABeta [0,1)
- OABeta (0,1]
- ZOABeta [0,1]

is on the ToDo already.

**Comment by Pranavesh-Panakkal** [comment]
Hi, I see ZABeta() is already implemented. Could you give some guidance on how to model ZOIB (ZOABeta)? Thanks.
